app/assets/javascripts/components/article_finder/article_finder.jsx

```diff
@@ -52,20 +55,40 @@ const ArticleFinder = createReactClass({
     }
   },
 
+  getParamsURL() {
+    const query = qs.parse(window.location.search);
+    const entries = Object.entries(query);
+    entries.map(([key, val]) => {
+      val = (key === 'article_quality') ? parseInt(val) : val;
+      return this.updateFields(key, val);
+    });
+  },
   updateFields(key, value) {
-    return this.props.updateFields(key, value);
+    const update_field = this.props.updateFields(key, value);
+    Promise.resolve(update_field).then(() => {
+    if (this.props.search_term.length !== 0) { this.buildURL(); }
+    });
   },
 
   toggleFilter() {
     return this.setState({
       showFilters: !this.state.showFilters
     });
   },
-
+  buildURL() {
+    let queryStringUrl = window.location.href.split('?')[0];
+    const params_array = ['search_type', 'article_quality', 'min_views'];
+    queryStringUrl += `?search_term=${this.props.search_term}`;
+    params_array.map((param) => {
+      return queryStringUrl += `&${param}=${this.props[param]}`;
+    });
```

Comment:
In this case, you're really just using `.map()` as a `.forEach()`. You can change it to a `.forEach()` or we could use `.reduce()`. I think the following would work:

```js
queryStringUrl = params_array.reduce((url, param) => {
  return url + `&${param}=${this.props[param]}`;
}, queryStringUrl);
```

I don't really have a preference! In fact, I think @ragesoss might like the `.forEach()` better.  :)

Reply:
Heh. `.reduce()` is fine here. But I probably would have used `.forEach()` instead if I'd written it myself, yeah.

Reply:
Let's go with .forEach() ! :)